### PR TITLE
Dismiss completion upon Select-All

### LINKED ISSUE #1659: Select All doesn't close IntelliSense
Pressing Ctrl+A when the cursor is at the end of the text buffer, then typing over the entire document doesn't close any active IntelliSense window, causing it to show incorrect completions for the new text.

This is particularly annoying (and likely to happen) in the Watch window, with keystroke sequences like `int.<Ctrl+A>t<Tab>`

### PR DESCRIPTION
**Pull request by rchande**
Fixes #1659 

Potential reviewers: @balajikris @basoundr @brettfo @dpoeschl @jasonmalinowski  @Pilchie

### CODE REVIEW

**Comment by rchande** [comment]
Working on an integration test.

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_SelectAll.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+using Microsoft.CodeAnalysis.Editor.Commands;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.IntelliSense.Completion
+{
+    internal partial class Controller
+    {
+        CommandState ICommandHandler<SelectAllCommandArgs>.GetCommandState(SelectAllCommandArgs args, System.Func<CommandState> nextHandler)
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_SelectAll.cs]: Unnecessary "System." qualification on Func.

**Comment by Pilchie** [comment]
:+1:

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_SelectAll.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            AssertIsForeground();
+            return nextHandler();
+        }
+
+        void ICommandHandler<SelectAllCommandArgs>.ExecuteCommand(SelectAllCommandArgs args, Action nextHandler)
+        {
+            AssertIsForeground();
+            if (sessionOpt != null)
+            {
+                StopModelComputation();
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_SelectAll.cs]: DismissSessionIfActive();?  Does it matter? "StopModelComputation" sounds like completion will continue if the model is already computed.

File: src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb [hunk 3]
```diff
@@ -1485,5 +1485,24 @@ class Program
...
+class C
+{
+    void foo(int x)
+    {
+        $$]]></Document>)
+                ' Note: the caret is at the file, so the Select All command's movement
+                ' of the caret to the end of the selection isn't responsible for 
+                ' dismissing the session.
+                state.SendInvokeCompletionList()
+                state.SendSelectAll()
```
Comment (dpoeschl) [inline src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb]: Should we assert that something is in the list first?

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_SelectAll.cs (hunk 2 above)
Comment (rchande) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_SelectAll.cs]: Guess what "StopModelComputation" calls and how it decides if it should do anything.

But yeah, this should probably be consistent.

File: src/EditorFeatures/Test2/IntelliSense/TestState.vb [hunk 4]
```diff
@@ -233,6 +233,11 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.IntelliSense
...
         End Sub
 
+        Public Overloads Sub SendSelectAll()
+            Dim handler = DirectCast(CompletionCommandHandler, ICommandHandler(Of SelectAllCommandArgs))
+            MyBase.SendSelectAll(Sub(a, n) handler.ExecuteCommand(a, n), Sub() Return)
+        End Sub
+
         Public Sub AssertNoCompletionSession(Optional block As Boolean = True)
             If block Then
                 WaitForAsynchronousOperations()
```
Comment (Pilchie) [inline src/EditorFeatures/Test2/IntelliSense/TestState.vb]: Should we have a test for VB too?

**Comment by rchande** [comment]
Fixes #1659

**Pull request merged**